shell/platform/android/io/flutter/view/AccessibilityViewEmbedder.java

```diff
@@ -0,0 +1,312 @@
+        AccessibilityNodeProvider provider = origin.view.getAccessibilityNodeProvider();
+        if (provider == null) {
+            // The provider is null for views that don't have a virtual accessibility tree.
+            // We currently only support embedding virtual hierarchies in the Flutter tree.
+            // TODO(amirh): support embedding non virtual hierarchies.
+            // https://github.com/flutter/flutter/issues/29717
+            return null;
+        }
+        AccessibilityNodeInfo originNode =
+                origin.view.getAccessibilityNodeProvider().createAccessibilityNodeInfo(origin.id);
+        return convertToFlutterNode(originNode, flutterId, origin.view);
+    }
+
+    private AccessibilityNodeInfo convertToFlutterNode(AccessibilityNodeInfo originNode, int flutterId, View embeddedView) {
+        AccessibilityNodeInfo result = AccessibilityNodeInfo.obtain(rootAccessibilityView, flutterId);
+        result.setPackageName(rootAccessibilityView.getContext().getPackageName());
+        result.setSource(rootAccessibilityView, flutterId);
+        result.setClassName(originNode.getClassName());
+
+        Rect displayBounds = flutterIdToDisplayBounds.get(flutterId);
+
+        copyAccessibilityFields(originNode, result);
+        translateBounds(originNode, displayBounds, result);
+        addChildren(originNode, embeddedView, displayBounds, result);
+        setParent(originNode, embeddedView, result);
+
+        return result;
+    }
+
+    private void setParent(AccessibilityNodeInfo originNode, View embeddedView, AccessibilityNodeInfo result) {
+        Long parentOriginPackedId = reflectionAccessors.getParentNodeId(originNode);
+        if (parentOriginPackedId == null) {
+            return;
+        }
+        int parentOriginId = ReflectionAccessors.getVirtualNodeId(parentOriginPackedId);
+        int parentFlutterId = originToFlutterId.get(new ViewAndId(embeddedView, parentOriginId));
+        result.setParent(rootAccessibilityView, parentFlutterId);
+    }
+
+
+    private void addChildren(AccessibilityNodeInfo originNode, View embeddedView, Rect displayBounds, AccessibilityNodeInfo resultNode) {
+        for (int i = 0; i < originNode.getChildCount(); i++) {
+            Long originPackedId = reflectionAccessors.getChildId(originNode, i);
+            if (originPackedId == null) {
+                continue;
+            }
+            int originId = ReflectionAccessors.getVirtualNodeId(originPackedId);
+            ViewAndId origin = new ViewAndId(embeddedView, originId);
+            int childFlutterId;
+            if(originToFlutterId.containsKey(origin)) {
+                childFlutterId = originToFlutterId.get(origin);
+            } else {
+                childFlutterId = nextFlutterId++;
+                originToFlutterId.put(origin, childFlutterId);
+                flutterIdToOrigin.put(childFlutterId, origin);
+                flutterIdToDisplayBounds.put(childFlutterId, displayBounds);
```

Comment:
Do we need to worry display bounds get out of sync frequently?

Reply:
Interesting, I think the following scenario is be possible:

  * platform view resized on UI thread
  * platform view resized on Platform thread
  * [UI thread sends the new semantics tree] createAccessibilityNodeInfos is called

This will result in the platform view's a11y tree being ahead of the AccessibilityBridge's semantics tree. Not sure what to do about it though... it will be very brief anyway...

Reply:
As long as changes in the nodes from the virtual tree still prompt talkback to ask for more nodes, I don't think they would stay out of sync for more than a frame or two. We're already a few frames behind for flutter, so that is fine